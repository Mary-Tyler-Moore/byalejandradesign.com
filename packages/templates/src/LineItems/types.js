/** @flow */
export type Props = {
  lineItems: Array<{
    name: string,
    description: string,
    quantity: number,
    unitAmount: number,
    totalAmount: number,
    kind: 'debit' | 'credit',
  }>,
};
