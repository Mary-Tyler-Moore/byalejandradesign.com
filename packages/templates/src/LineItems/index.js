/** @flow */
export { default } from './LineItems';
export * from './types';
