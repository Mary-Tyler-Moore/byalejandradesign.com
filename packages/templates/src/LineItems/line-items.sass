@import ~@njmyers/sass/material/typography

+typography("sourceSans", ("Source Sans Pro", sans-serif), 100, 300, 600, 800)

$border-color: transparentize(black, 0.85)

.lineItems
  margin: 0
  width: 100%
  border-collapse: collapse
  border-spacing: 0px
  table-layout: auto
  padding: 0

  &_table
    margin-top: 0.75rem
    width: 100%

  &_thead

  &_th
    width: 20%
    text-align: left

    @extend %subtitle-sourceSans-2

    padding: 1.25rem 0
    border-bottom: 1px solid $border-color
    border-top: 1px solid $border-color

    &-first
      width: 40%

  &_tbody

  &_tr

  &_td
    @extend %body-sourceSans-2

    padding: 1.25rem 0
    text-align: left
    border-bottom: 1px solid $border-color
