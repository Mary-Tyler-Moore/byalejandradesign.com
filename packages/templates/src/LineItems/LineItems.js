/** @flow */
import * as React from 'react';
import Line from '../Line';
import { kebabToTitle, dollarString } from 'smalldash';
// types
import type { Props } from './types';
// styles
import './line-items.sass';

class LineItems extends React.Component<Props> {
  render() {
    return (
      Array.isArray(this.props.lineItems) &&
      this.props.lineItems.length && (
        <section className="lineItems">
          <h5 className="h5-amiri">Items</h5>
          <ul className="lineItems_ul">
            {this.props.lineItems.map((item) => (
              <li className="lineItems_li" key={item.name}>
                <h6 className="h6-amiri">{kebabToTitle(item.name)}</h6>
                <Line label="quantity">{item.quantity}</Line>
                <Line label="price">{dollarString(item.unitAmount)}</Line>
                <Line label="totalPrice">{dollarString(item.totalAmount)}</Line>
              </li>
            ))}
          </ul>
        </section>
      )
    );
  }
}

export default LineItems;
