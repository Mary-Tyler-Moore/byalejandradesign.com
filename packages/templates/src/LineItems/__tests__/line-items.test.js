import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import LineItems from '../LineItems';

const item = {
  name: 'shoreline-cups',
  description:
    'Yes, they are smaller than your regular size. My husband loves drinking tea/coffee in these guys. He says they are the perfect ',
  quantity: 1,
  unitAmount: 15,
  totalAmount: 15,
  kind: 'debit',
};

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<LineItems />);
    expect(render.toJSON()).toMatchSnapshot();
  });

  test('it loads line items', () => {
    const render = TestRenderer.create(<LineItems lineItems={[item]} />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
