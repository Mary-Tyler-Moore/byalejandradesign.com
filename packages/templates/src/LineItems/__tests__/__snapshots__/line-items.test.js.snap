// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads line items 1`] = `
<section
  className="lineItems"
>
  <h5
    className="h5-amiri"
  >
    Items
  </h5>
  <ul
    className="lineItems_ul"
  >
    <li
      className="lineItems_li"
    >
      <h6
        className="h6-amiri"
      >
        Shoreline Cups
      </h6>
      <p
        className="body-sourceSans-2"
      >
        <strong>
          Quantity
          : 
        </strong>
        1
      </p>
      <p
        className="body-sourceSans-2"
      >
        <strong>
          Price
          : 
        </strong>
        $15.00
      </p>
      <p
        className="body-sourceSans-2"
      >
        <strong>
          Total Price
          : 
        </strong>
        $15.00
      </p>
    </li>
  </ul>
</section>
`;

exports[`it works it loads without crashing 1`] = `null`;
